Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -337,8 +337,13 @@ extension TabScrollingController: UIScrollViewDelegate {
     }
 
     func scrollViewShouldScrollToTop(_ scrollView: UIScrollView) -> Bool {
-        showToolbars(animated: true)
-        webViewContainerToolbar?.isHidden = false
+        if let webViewContainerToolbar = webViewContainerToolbar {
+            if webViewContainerToolbar.isHidden {
```

Comment:
I think using the webViewContainerToolbar here is a bit misleading. `webViewContainerToolbar` doesnt actually repersent the main toolbars. (sorry the name _is_ confusing.) instead I would try to use `toolbarState == .collapsed` instead. 

```
if toolbarState == .collapsed {   
  showToolbars()
  webViewContainerToolbar?.isHidden = false
  return false
}
```

Reply:
Oh, I hadn't seen that flag but that does make more sense. I'll submit those changes and have a look at updating the tests within a day 👍 Thanks for welcoming me, and taking the time to give suggestions!